src/cmd/services/m3coordinator/downsample/metrics_appender.go

```diff
@@ -327,6 +339,132 @@ func (a *metricsAppender) Finalize() {
 	a.tagEncoder = nil
 }
 
+func (a *metricsAppender) newSamplesAppenders(
+	stagedMetadata metadata.StagedMetadata,
+) ([]samplesAppender, error) {
+	var (
+		pipelines []metadata.PipelineMetadata
+		appenders []samplesAppender
+	)
+	for _, pipeline := range stagedMetadata.Pipelines {
+		// For pipeline which have tags to augment we generate and send
+		// separate IDs. Other pipelines return the same.
+		pipeline := pipeline
```

Comment:
Hm, is this needed to capture the var or something?

Reply:
not needed currently..